astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,380 @@
+
+    return hcopy
+
+
+def time_to_fits(table):
+    """
+    Replace Time columns in a Table with non-mixin columns containing
+    each element as a vector of two doubles (jd1, jd2) and return a FITS
+    header with appropriate time coordinate keywords.
+    jd = jd1 + jd2 represents time in the Julian Date format with
+    high-precision.
+
+    Parameters
+    ----------
+    table : `~astropy.table.Table`
+        The table whose Time columns are to be replaced.
+
+    Returns
+    -------
+    table : `~astropy.table.Table`
+        The table with replaced Time columns
+    hdr : `~astropy.io.fits.header.Header`
+        Header containing Cards associated with the FITS time coordinate
+    """
+
+    # Shallow copy of the input table
+    newtable = table.copy(copy_data=False)
+
+    # Global time coordinate frame keywords
+    hdr = Header([Card(keyword=key, value=val[0], comment=val[1])
+                  for key, val in GLOBAL_TIME_INFO.items()])
+
+    time_cols = table.columns.isinstance(Time)
+
+    # Geocentric Position
+    pos_geo = None
+
+    for col in time_cols:
+        # The following is necessary to deal with multi-dimensional ``Time`` objects
+        # (i.e. where Time.shape is non-trivial).
+        jd12 = np.array([col.jd1, col.jd2])
+        # Roll the 0th (innermost) axis backwards, until it lies in the last position
+        # (jd12.ndim)
+        jd12 = np.rollaxis(jd12, 0, jd12.ndim)
+        newtable.replace_column(col.info.name, Column(jd12, unit='d'))
+
+        # Get column position(index)
+        n = table.colnames.index(col.info.name) + 1
+
+        # Time column override keywords
+        hdr.append(Card(keyword='TCTYP{}'.format(n), value=col.scale.upper()))
+
+        # Time column reference positions
+        if col.location is None:
+            if pos_geo is not None:
+                warnings.warn(
+                    'Time Column "{}" has no specified location, but global Time Position '
+                    'is present, which will be the default for this column in '
+                    'FITS specification.'.format(col.info.name),
+                    AstropyUserWarning)
+        else:
+            hdr.append(Card(keyword='TRPOS{}'.format(n), value='TOPOCENTER'))
+            # Compatibility of Time Scales and Reference Positions
+            if col.scale in BARYCENTRIC_SCALES:
+                warnings.warn(
+                    'Earth Location "TOPOCENTER" for Time Column "{}" is incompatabile '
+                    'with scale "{}".'.format(col.info.name, col.scale.upper()),
+                    AstropyUserWarning)
+            if col.location.size > 1:
+                raise ValueError('Vectorized Location of Time Column "{}" cannot be written, '
+                                 'as it is not supported.'.format(col.info.name))
+            if pos_geo is None:
+                hdr.extend([Card(keyword='OBSGEO-'+ dim.upper(), value=getattr(col.location, dim).value)
```

Comment:
The end bit of this should be `to_value(u.m)` - otherwise, the unit might well be wrong. (Do add test case with `EarthLocation` in units of `km`.)